src/bucket/BucketListSnapshot.h

```diff
@@ -93,10 +81,24 @@ class SearchableBucketListSnapshot : public NonMovableOrCopyable
 
     std::shared_ptr<LedgerEntry> getLedgerEntry(LedgerKey const& k);
 
+    // Loads LedgerEntry from the specified historical snapshot. Returns
+    // <LedgerEntryPtr, true> if the snapshot for the given ledger is available,
+    // <nullptr, false> otherwise. Note that ledgerSeq is defined as the state
+    // of the BucketList at the beginning of the ledger. This means that for
+    // ledger N, the maximum lastModifiedLedgerSeq of any LedgerEntry in the
+    // BucketList is N - 1.
+    std::pair<std::shared_ptr<LedgerEntry>, bool>
```

Comment:
Aren't you emulating std::optional here?

Reply:
Yes, but I'd rather avoid wrapping a pointer in an optional. We recently had a segfault bug in the TxFrame refactor where a pointer was wrapped in an optional, but only the optional was null checked, not the underlying pointer. It was tricky to track down and made the code confusing, so I'd rather keep the pair as is.